src/us/kbase/workspace/kbase/SchemaUpdaterCLI.java

```diff
@@ -0,0 +1,149 @@
+import java.util.concurrent.Callable;
+
+import org.slf4j.LoggerFactory;
+
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoDatabase;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParameterException;
+import picocli.CommandLine.Parameters;
+import picocli.CommandLine.Spec;
+import us.kbase.workspace.database.mongo.SchemaUpdater;
+import us.kbase.workspace.database.mongo.SchemaUpdater.SchemaUpdateException;
+import us.kbase.workspace.kbase.InitWorkspaceServer.WorkspaceInitException;
+import us.kbase.workspace.kbase.KBaseWorkspaceConfig.KBaseWorkspaceConfigException;
+
+/** A CLI class to update the workspace schema. */
+@Command(
+		name = "update_workspace_database_schema",
+		mixinStandardHelpOptions = true,
+		version = VERSION,
+		description = "Update the workspace database to the current version.\n" +
+				"Please read the upgrade documentation carefully before running this script."
+		)
+public class SchemaUpdaterCLI implements Callable<Integer>{
+	
+	// don't support the KB_SERVICE_NAME env var garbage
+	private final static String CONFIG_SECTION = "Workspace";
+	
+	@Spec
+	private CommandSpec spec;
+	
+	@Parameters(
+			index = "0",
+			paramLabel = "<CONFIG_FILE>",
+			description = "The configuration file (usually called deploy.cfg) for the workspace " +
+					"service to modify.")
+	private Path configFile;
+	
+	@Option(names = {"-c", "--complete"},
+			description = "Complete the upgrade. WARNING: DO NOT complete the upgrade unless " +
+				"no other processes are writing to the workspace database.")
+	private boolean complete;
+	
+	@Option(names = {"-o", "--override-version-check"},
+			description = "Allow the upgrade to continue even if the database schema version " +
+				"indicates it is complete.")
+	private boolean override;
+	
+	static {
+		// mongo sure is chatty
+		((Logger) LoggerFactory.getLogger("org.mongodb.driver")).setLevel(Level.ERROR);
+	}
+
+	/** Provides a MongoClient given a configuration. */ 
+	public static interface MongoProvider {
+		
+		/** Provide the MongoClient.
+		 * @param cfg the configuration to use to build the client.
+		 * @return the client.
+		 * @throws WorkspaceInitException if the client couldn't be created.
+		 */
+		public MongoClient provide(final KBaseWorkspaceConfig cfg) throws WorkspaceInitException;
+	}
+
+	private final SchemaUpdater updater;
+	private final MongoProvider mongoProvider;
+	
+	private SchemaUpdaterCLI(final SchemaUpdater updater, final MongoProvider mongoProvider) {
+		this.updater = updater;
+		this.mongoProvider = mongoProvider;
+	}
+
+	/** Run the CLI.
+	 * @param args the CLI arguments.
+	 */
+	public static void main(final String[] args) {
+		// WARNING - these lines are only tested manually. Retest if you make changes.
```

Comment:
⚠️ ❗ ☠️  😱

Reply:
Pretty hard to test a function that `System.exit()`s